sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -286,7 +287,9 @@ private[hive] class SparkExecuteStatementOperation(
           if (e.isInstanceOf[HiveSQLException]) {
             throw e.asInstanceOf[HiveSQLException]
           } else {
-            throw new HiveSQLException("Error running query: " + e.toString, e)
+            val root = ExceptionUtils.getRootCause(e)
```

Comment:
Could we change it to?
```scala
setState(OperationState.ERROR)
e match {
  case hiveException: HiveSQLException =>
    logError(s"Error executing query with $statementId, currentState $currentState, ", e)
    HiveThriftServer2.listener.onStatementError(
      statementId, hiveException.getMessage, SparkUtils.exceptionString(hiveException))
    throw hiveException
  case _ =>
    val rootCause = Option(ExceptionUtils.getRootCause(e)).getOrElse(e)
    logError(
      s"Error executing query with $statementId, currentState $currentState, ", rootCause)
    HiveThriftServer2.listener.onStatementError(
      statementId, rootCause.getMessage, SparkUtils.exceptionString(rootCause))
    throw new HiveSQLException("Error running query: " + rootCause.toString, rootCause)
}
```

Reply:
> ```scala
> val rootCause = Option(ExceptionUtils.getRootCause(e)).getOrElse(e)
> ```

Return null only if the input `e` is null. Do we still add this option?

Reply:
Besides the null checker, I've changed the code to above style. @wangyum